frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx

```diff
@@ -73,16 +73,21 @@ export const AddCapacityModal = withHandlePromise((props: AddCapacityModalProps)
       path: `/spec/storageDeviceSets/0/count`,
       value: presentCount + 1,
     };
-    props
-      .handlePromise(k8sPatch(OCSServiceModel, ocsConfig, [patch]))
-      .then(() => {
-        setProgress(false);
-        close();
-      })
-      .catch((error) => {
-        setError(error);
-        setProgress(false);
-      });
+    if (!storageClass) {
+      setError('No StorageClass selected');
+      setProgress(false);
+    } else {
+      props
+        .handlePromise(k8sPatch(OCSServiceModel, ocsConfig, [patch]))
```

Comment:
```suggestion
        .k8sPatch(OCSServiceModel, ocsConfig, [patch]))
```
No need to use `handlePromise` since the props provided by it is not used anywhere in the code.